examples/handlebars-ad-renderer/src/MyPluginImpl.ts

```diff
@@ -1,10 +1,37 @@
 import { core, extra } from "@mediarithmics/plugins-nodejs-sdk";
 
+export interface MyInstanceContext extends core.AdRendererRecoTemplateInstanceContext {
+  render_template?: (...args: any[]) => string;
+}
+
 export class MyHandlebarsAdRenderer extends core.AdRendererRecoTemplatePlugin {
+
+  protected async instanceContextBuilder(creativeId: string, forceReload = false): Promise<MyInstanceContext> {
+
+    const baseInstanceContext = await super.instanceContextBuilder(creativeId);
+
+    const templateProp = baseInstanceContext.properties.findDataFileProperty("template");
```

Comment:
Does it mean that the plugin property of the adrenderer should always be called template ?

Reply:
The code you're referring to is in the "Plugin Implementation" which is the responsibility of the Plugin dev.

This code is present in the SDK codebase only as an example and is part of the SDK testing => the plugin dev can follow or not the example, and they are free to use another property to store their template.

However, note that there is other part of the Plugin SDK that are making assumptions over how the Plugin Properties are named. The default impl. (which can be completely overridden by the plugin dev and is therefore optional) of `AdRendererTemplatePlugin.instanceContextBuilder()` is a good example here: https://github.com/MEDIARITHMICS/plugins-nodejs-sdk/pull/61/files#diff-3794584533e332ac1ad5cb8111495ce0R29

Those part are always optional as the SDK should be usable in a way that makes no assumption at all on how the Plugin properties are organised. But we're still proposing them for Plugin dev. that prefers to use a "predefined" Plugin Properties schema and rely on SDK code rather than managing properties's logic themselves.